### PR TITLE
feat(#941): write a portable handoff automatically so the usage-limit breadcrumb points at something

### LINKED ISSUE #941: Portable handoff exists only when /pause is invoked, so the usage-limit breadcrumb points at nothing
## Problem

When a session dies on an Anthropic usage limit, `usage-limit-record.sh` writes a breadcrumb — and that breadcrumb is supposed to point at the most recent portable handoff so the next session reads *what was happening* instead of reconstructing it from a transcript.

On 2026-08-01 the account hit its 5-hour window and killed three concurrent sessions plus their subagents (9 records between 23:10 and 23:29 ET, across `inventory`, `claude-code-config`, and `festive-wright`). Every record came out with:

```json
"portable_handoff": null
```

There were **zero** `portable-handoff-*.md` files on disk. The pointer degraded to transcript-only in exactly the situation it was built for.

## Why

`/pause` is the only thing that writes that document, and by contract it "runs because you asked, and for no other reason" (`.claude/skills/pause/SKILL.md`). That contract is correct and should not change — but it means the artifact exists only when a human saw the wall coming. A limit that arrives without warning is the common case, and it is the case with no document.

This is not the pre-emptive wind-down blocked upstream (#824 / #835). Nothing here detects, estimates, or predicts anything. It only makes sure the artifact the recorder already looks for is actually on disk when it looks.

## Proposed change

Add a **document-only checkpoint**: the handoff artifact, written automatically, with none of `/pause`'s wind-down.

Split the two things `/pause` currently fuses:

| | Stops work | Writes the document | Trigger |
|---|---|---|---|
| `/pause` (unchanged) | yes | yes, model-rendered | human asks |
| checkpoint (new) | **no** | yes, mechanically collected | `SubagentStop` hook |

- **New script `.claude/scripts/checkpoint-handoff.sh`** — collects the mechanical state the template needs (working directory, branch, uncommitted/unpushed state, open PRs, in-flight agents from `session-state.json`) and renders `portable-handoff-<stamp>-<session>-checkpoint.md` into `~/.claude/handoffs/`. Same filename glob the recorder already scans, same atomic `mktemp` + `mv` publish, validated by `portable-handoff-lint.sh` before it ships.
- **Back-pointer, so freshness never destroys depth.** The recorder takes newest-by-mtime, so a checkpoint will usually outrank an older model-rendered `/pause` document. The checkpoint therefore names the most recent non-checkpoint handoff in its own body — the reader gets current git state *and* the last real narrative.
- **Throttle.** Rewrite at most once per `CLAUDE_CHECKPOINT_MIN_INTERVAL` (default 10 min), and skip entirely when nothing observable changed since the last one. A checkpoint that fires on every subagent exit is its own kind of waste.
- **Retention.** Prune `*-checkpoint.md` older than 7 days (matching the recorder's `HANDOFF_MAX_AGE_DAYS`). Never prune a `/pause`-authored document — those are the user's.
- **Fail-soft, always exit 0.** A checkpoint that becomes a new failure mode is worse than no checkpoint, for the same reason the recorder is written that way.

## Non-goals

- No token, spend, or quota arithmetic anywhere in this change. `.claude/rules/safety.md` §"Anthropic Quota & Spend Authority" is satisfied as written and is **not** amended.
- No gating: this never pauses, downgrades, defers, or refuses work.
- `/pause` keeps its current behavior and its "does not fire on a schedule, a threshold, or an inference" clause — the checkpoint is a different artifact producer, not `/pause` on a timer.
- Not the pre-emptive wind-down. #835 still tracks that, and when a real signal appears it emits this same artifact.

## Acceptance criteria

- [ ] `.claude/scripts/checkpoint-handoff.sh` exists, is executable, exits 0 on every path including missing `jq`, missing `session-state.json`, and a non-repo cwd
- [ ] It writes `~/.claude/handoffs/portable-handoff-<stamp>-<session>-checkpoint.md`, matching the glob `usage-limit-record.sh` scans
- [ ] Output passes `portable-handoff-lint.sh` (all required sections present and non-empty; no harness paths, phase vocabulary, state-file names, skill invocations, or unrendered placeholders)
- [ ] The document names an absolute working directory, per the lint's `working-directory-absolute` rule
- [ ] When a non-checkpoint handoff exists, the checkpoint body names its path
- [ ] Throttled: a second invocation inside the min-interval with unchanged state writes nothing and exits 0
- [ ] Retention prunes only `*-checkpoint.md` older than 7 days; a `/pause`-authored document of any age is never removed
- [ ] Registered as a `SubagentStop` hook with a bounded timeout
- [ ] `usage-limit-record.sh` needs no change — verified by a test that runs it against a directory containing a checkpoint and asserts `portable_handoff` is non-null
- [ ] Test suite under `.claude/scripts/tests/` covering the above, passing
- [ ] `.claude/skills/pause/SKILL.md` gains a short pointer distinguishing the two artifacts; `/pause`'s own steps are unchanged
- [ ] Rule corpus word count unchanged or justified (this is reference/script work, not rule work)

## Context

- `.claude/reference/usage-limit-signal-audit-2026-07.md` — why no pre-emptive trigger exists
- `.claude/hooks/usage-limit-record.sh` — the consumer of this artifact
- `.claude/skills/pause/references/portable-handoff-template.md` — the document contract
- `.claude/scripts/portable-handoff-lint.sh` — the portability gate
- Related: #824 (audit), #835 (re-audit + wind-down when a signal appears), #499 (the `/quota` rollback that produced the no-estimation rule)

---

## Implementation Plan

_Source: Claude's analysis. CodeRabbit's plan was requested by CI (`@coderabbitai plan`) but did not arrive within the 5-minute poll window — CR appears rate-limited following the 2026-08-01 account limit. Proceeding without it per the issue-planning flow._

### The constraint that shapes the design

`portable-handoff-lint.sh` rejects any relative `.claude/` path (`RE_HARNESS_PATH='\.claude/'`), and the only exemption is an **absolute** path containing `/.claude/worktrees/`. It also rejects phase vocabulary, state-file names, and any `/skill-name` invocation resolved from the live skill catalog.

A naive mechanical renderer would therefore fail its own gate the moment it runs in this repository: the uncommitted-file list is full of `.claude/scripts/…`, and PR titles routinely contain slash-command names. This is a real tension the template's prose rendering rules never had to confront, because a human-driven renderer rewrites offending lines by hand.

**Resolution: a bounded degrade ladder.** Render, lint, and on failure drop to the next-less-specific rendering. Ship the first tier that passes; write nothing if none does.

| Tier | Contents |
|---|---|
| 1 | Full — changed-file paths listed, pull request and issue titles quoted |
| 2 | File paths replaced by a count plus a `git status` instruction |
| 3 | Titles dropped; pull requests and issues named by number and URL only |
| 4 | Minimal skeleton — branch, absolute working directory, counts, universal commands |

At most four renders and four lint calls, all local. This makes "output passes the lint" a property the script *enforces on itself* rather than one the author hopes holds in every repository it ever runs in.

### Files

| Path | Change |
|---|---|
| `.claude/scripts/checkpoint-handoff.sh` | new — collector + renderer + degrade ladder + throttle + retention |
| `global-settings.json` | new `SubagentStop` entry (auto-registered into `~/.claude/settings.json` by `session-start-sync.sh` → `register-hooks.py`) |
| `.claude/scripts/tests/checkpoint-handoff.test.sh` | new — suite covering every acceptance criterion |
| `.claude/hooks/README.md` | document the new hook |
| `.claude/skills/pause/SKILL.md` | short pointer distinguishing the two artifacts; steps unchanged |

### Script behavior

1. **Guard.** Require `jq` and `git`; exit 0 if absent. Never fail the turn.
2. **Throttle.** Skip when the newest existing `*-checkpoint.md` is younger than `CLAUDE_CHECKPOINT_MIN_INTERVAL` (default 600s) **and** the state fingerprint (branch, HEAD SHA, `git status --porcelain` digest, unpushed count) is unchanged. Fingerprint stored in the document as an HTML comment so no sidecar state file is needed — and so nothing new needs a write lock.
3. **Collect.** Branch, absolute working directory, `git status --porcelain`, `origin/<branch>..HEAD` count, open pull requests authored by the user (best-effort `gh`, bounded, skipped on failure), and the issue number parsed from an `issue-N-*` branch.
4. **Back-pointer.** Newest `portable-handoff-*.md` that is *not* a checkpoint, named in the body so a fresh shallow checkpoint never buries an older rich one.
5. **Render + lint ladder** as above, into `$OUT_DIR` via `mktemp` so the publish is a same-filesystem atomic `mv`, matching `/pause`.
6. **Retention.** Delete `*-checkpoint.md` older than 7 days. `find -name '*-checkpoint.md'` only — a `/pause`-authored document is never a deletion candidate at any age.
7. **Exit 0 on every path.**

### Rendering the required sections mechanically

- **Start here** — imperative, universal: run `git status` in the named directory, then `gh pr view <N>` for anything in flight.
- **What we're working on** — repository, branch, and the issue number and title parsed from the branch name; explicit "written automatically, so this is what the branch says, not a statement of intent."
- **Open work** — one block per open pull request, or "Nothing is in flight."
- **Decisions made this session** — honestly empty: "This checkpoint was written automatically and records no decisions," plus the back-pointer when one exists. Non-empty, so `required-sections` passes without pretending.
- **Local state on this machine** — absolute working directory (the one absolute path the lint requires), branch, change counts.

### Safety

No token, spend, or quota arithmetic. No gating. Nothing pauses, downgrades, defers, or refuses work. `.claude/rules/safety.md` §"Anthropic Quota & Spend Authority" holds unmodified, as it did for the recorder.

### Rule corpus

Zero words added to `CLAUDE.md` or `.claude/rules/*.md`. Everything here is script, test, and reference-tier documentation.

### PR DESCRIPTION
**Pull request by auerbachb**
## What this does

When a session dies on an Anthropic usage limit, `usage-limit-record.sh` writes a breadcrumb naming the most recent portable handoff — so the next session reads *what was happening* instead of reconstructing it from a transcript.

On 2026-08-01 the account limit killed three concurrent sessions and their subagents (9 records, 23:10–23:29 ET, across three repos). Every record came out with:

```json
"portable_handoff": null
```

Zero `portable-handoff-*.md` files existed. The only thing that writes one is invoked by hand, and a limit that arrives without warning is exactly the case where nobody did.

This adds `.claude/hooks/checkpoint-handoff.sh`, registered on `SubagentStop`: it writes the **artifact** and nothing else — no wind-down, no stopping, no decision. `/pause` keeps its "runs because you asked, and for no other reason" contract, because this is a different producer of the same document, not that command on a timer.

Not the pre-emptive wind-down — that stays blocked upstream (#824 / #835), and the runtime is still 2.1.219, the exact build that audit covers.

## The degrade ladder

`portable-handoff-lint.sh` rejects relative `.claude/` paths, phase vocabulary, state-file names, and slash-command names. A mechanical renderer walks into all of them: in *this* repo the changed-file list is nothing but `.claude/scripts/…`. A human rewrites the offending line; a script cannot.

So it renders, lints, and drops a tier on failure, shipping the first that passes — and writing **nothing** if none does:

| Tier | Contents | |
|---|---|---|
| 1 | file paths listed, PR titles quoted | verified to fail here with the exact `[harness-path]` violation |
| 2 | count + `git status` instruction | the normal result in this repo |
| 3 | titles dropped, number + URL only | |
| 4 | minimal — absolute working directory, counts | |

This makes "the output is portable" a property the script enforces on itself in whatever repo it runs in, not one I hoped would hold everywhere.

## Bug found and fixed in self-review

PR fields were joined with `@tsv` and read with `IFS=$'\t'`. Tab is IFS **whitespace**, so `read` collapses runs of it — a titleless pull request became `number=42, title=<the URL>, url=<empty>`, rendering a URL where a title belongs with **nothing failing anywhere** (the result is lint-clean). Now joined on US (`0x1f`), which is not IFS whitespace.

Paired control against a copy differing only in those two lines:

```
pre-fix   - **Pull request 42 — https://example.com/42**
            <blank URL line>
post-fix  - **Pull request 42**
            https://example.com/42
```

T9 pins both directions, so it fails against the pre-fix code.

## Review coverage: `none` — both local CLIs down

Per `cr-local-review.md`, recorded here rather than glossed:

- **CodeAnt — failed run.** `API Error: Access denied (403)` on stderr with `meta.capped: true` and 7 files skipped; the 15 files it did review were a stale-base set that did not include either file this PR adds. Not retried past once, and `codeant logout/login` deliberately not run (the 403 is a persistent account-wide entitlement, not auth).
- **CodeRabbit — dropped on timeout.** Reached `phase: analyzing / summarizing` and produced no further record for 10+ minutes, well past the 2-minute budget. Likely the same adaptive quota the account exhausted last night.

**Self-review ran instead** and is what produced the field-shift fix above. Self-review does not satisfy the merge gate — the GitHub reviewers still do.

## Verification

- `.claude/scripts/tests/checkpoint-handoff.test.sh` — 41 assertions, all passing
- Full CI-discovered bash suite: **64 suites, 0 failed**, with the new suite confirmed discovered (auto-discovery, no workflow edit)
- `shellcheck -x -S warning` clean. Ten `SC2016` *info* notices remain and are correct as written — they are literal backticks in generated Markdown; double-quoting those format strings would run `git status` as a command substitution. No suppression comments added.
- Rule corpus word count **unchanged** (11,719; cap 11,749) — script, test, and hook-README work only

A `T0` probe asserts the checker actually resolved before any "passes the portability check" assertion runs. Without it, every one of those could pass because no checker was found and the minimal tier shipped unverified — a pass for the wrong reason, and invisible.

## Test plan

- [ ] `.claude/hooks/checkpoint-handoff.sh` is executable and exits 0 on every path: missing `jq`, missing `git`, non-repo cwd, non-numeric settings, unknown option
- [ ] Writes `portable-handoff-<stamp>-<session>-checkpoint.md`, matching the glob `usage-limit-record.sh` already scans
- [ ] Output passes `portable-handoff-lint.sh` — verified in an ordinary repo (tier 1) and a harness-shaped one (tier 2)
- [ ] The document names an absolute working directory
- [ ] When a hand-written handoff exists, the checkpoint names it by basename (its absolute path is unportable and is deliberately not printed)
- [ ] An existing checkpoint is never offered as the hand-written handoff
- [ ] Throttle: unchanged state inside the window writes nothing; changed state inside the window still writes; elapsed window writes; `--force` overrides
- [ ] Retention prunes only `*-checkpoint.md` past the age bound — a hand-written handoff of the same age survives
- [ ] `usage-limit-record.sh` is unchanged and publishes a non-null `portable_handoff` when a checkpoint is present
- [ ] Registered on `SubagentStop` in `global-settings.json` with a bounded timeout, and `register-hooks.py` can resolve it (the script sits in `.claude/hooks/`, which is the only directory that registration joins manifest basenames onto)
- [ ] Empty PR title does not shift fields (T9, validated against the pre-fix code)
- [ ] `.claude/skills/pause/SKILL.md` gains a pointer distinguishing the two artifacts; its own steps are unchanged
- [ ] Rule corpus word count unchanged

Closes #941

___

## **CodeAnt-AI Description**
Automatically preserve portable session handoffs before unexpected usage-limit interruptions

### What Changed
- A background hook now creates a portable handoff while work is in progress, giving usage-limit records a real handoff to reference.
- Handoffs describe repository state, open pull requests, unfinished work, and running background units without stopping work or making decisions.
- Generated documents are checked for portability and simplified until they pass; if none pass, nothing is published.
- Checkpoints are throttled, retried when local state changes, retained for seven days, and kept separate from hand-written handoffs.
- Added coverage for portability fallback, throttling, publishing, retention, recorder integration, pull request formatting, and degraded environments.

### Impact
`✅ Fewer usage-limit records without a handoff`
`✅ Recoverable session state after unexpected interruptions`
`✅ Hand-written handoffs preserved`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

---

## Update — rebased onto `main`, adapted to PR #934

`main` advanced mid-work and one of the two commits was [PR #934](https://github.com/auerbachb/claude-code-config/pull/934), which added three portability rules: `open-work-ownership`, `pull-request-review-state`, `verification-command`. Every pull-request entry must now carry `Owner:`, `Waiting on:` and `Approval:`, and any document with in-flight work needs a `Verify with:`.

The renderer carried none of them, so the ladder degraded past every tier that lists pull requests and T9 failed on all four assertions. This is exactly the drift the rebase existed to catch — the ladder is only as verified as the checker it was verified against.

Each field is now **derived** from what the forge reports (author login, `reviewDecision`, `mergeStateStatus`) rather than emitted as a constant. The rules can only see presence — `Owner: yes` passes them — so a constant would satisfy the check and tell the reader nothing, which is the failure those rules were written against. Where the forge reports nothing, the line says so.

Merge state outranks review decision in `Waiting on:` — an approved branch that is `BEHIND` is waiting on the rebase, and saying it waits on review would point the reader at something that already happened. Verified against live data: PR #937 and PR #919 render `Approval: approved` with `Waiting on: the branch needs updating against the main branch first`.

Re-verified after the rebase: **48 assertions**, full discovered suite **66/66**, `shellcheck -S warning` clean, corpus still 11,719.

### Added acceptance criteria

- [ ] Every pull-request entry carries `Owner:`, `Waiting on:` and `Approval:`; the document carries at least one `Verify with:`
- [ ] Each is derived from real forge data, not a constant — asserted on values (per-entry owner, approved vs undecided), not just on the anchors
- [ ] `BEHIND` merge state outranks an existing approval in `Waiting on:`
- [ ] A pull request the forge reports no author/decision for still renders a truthful line rather than a guess

---

## Update — four review findings fixed (`5f04347`)

CodeAnt and Cursor each found two. All four were valid; all four are fixed, replied to, and resolved.

| Finding | Why it was real |
|---|---|
| `gh` failure rendered as "no open pull requests" (Cursor) | The exit status was discarded, so a failed query and an empty result were the same empty string. Reporting "nothing in flight" when the truth is "could not look" is the one mistake a handoff cannot afford — the reader has no way to detect it. Three states now: `ok` / `failed` / `skipped`. |
| `reissue-941` matched the issue regex (Cursor) | The boundary was `[^0-9]`, so any `re`/`de`/`un` prefix matched on its trailing letter and published a link to an unrelated issue. Now `[^[:alnum:]]`. |
| Fingerprint could not see staging (CodeAnt) | The status columns were stripped before hashing, so `git add` left the fingerprint unchanged and the throttle suppressed the checkpoint for the rest of the interval. Now hashes raw porcelain. |
| Same-second writes destroyed each other (CodeAnt) | Whole-second stamp + absent session id put two writers on one filename, and `mv -f` kept only the last. Not a rare interleave here — one account limit fails *every* session at once, which is exactly when concurrent writes happen. Now carries the `mktemp` suffix. |

**Partially declined, with reasoning:** the fingerprint still does not move when file *contents* change without the porcelain changing. Catching that means hashing the diff on every `SubagentStop`. It does not need catching, because time is the primary trigger — the next interval writes regardless, so the worst case is bounded at the interval rather than indefinite. The fingerprint only ever makes it write *more* often.

**Control for the subtle one:** under the old `sed 's/^...//'`, `?? new.txt` and `A  new.txt` both reduce to `new.txt` — byte-identical, so the pre-fix hash provably could not move. Raw porcelain differs. T10c pins it.

Re-verified: **58 assertions**, full discovered suite **66/66**, `shellcheck -S warning` clean.

### Postscript on local review coverage

The backgrounded CodeRabbit CLI run eventually returned, and its own record confirms what the coverage note above already claimed rather than softening it:

```json
{"type":"error","errorType":"timeout","message":"Review timed out (ID: aa63ea23-…)","recoverable":false}
```

Exit 1, `recoverable: false` — a failed run, not a silent clean pass. Coverage stays `none`; the GitHub reviewers are what actually covered this PR, and they found four real bugs.

### Added acceptance criteria

- [ ] A failed `gh` lookup renders as "could not be read", never as "none open"; a skipped lookup is a third distinct wording
- [ ] `reissue-941` publishes no issue link; `issue-941-real` still detects issue 941
- [ ] Staging a file moves the fingerprint
- [ ] Two same-second writes produce two distinct files and both survive; the unique name still matches the recorder's glob

### CODE REVIEW

**Comment by auerbachb** [comment]
@graphite-app re-review

File: .claude/hooks/checkpoint-handoff.sh [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+  TOPLEVEL=$(git rev-parse --show-toplevel 2>/dev/null)
+  BRANCH=$(git rev-parse --abbrev-ref HEAD 2>/dev/null)
+  HEAD_SHA=$(git rev-parse --short HEAD 2>/dev/null)
+  PORCELAIN=$(git status --porcelain 2>/dev/null)
+  if [[ -n "$PORCELAIN" ]]; then
+    CHANGED_COUNT=$(printf '%s\n' "$PORCELAIN" | grep -c . 2>/dev/null || true)
+    [[ "$CHANGED_COUNT" =~ ^[0-9]+$ ]] || CHANGED_COUNT=0
+    # Strip the two status columns; keep the path. Rename entries ("R  a -> b")
+    # keep both sides, which is what a reader wants to see anyway.
+    CHANGED_LIST=$(printf '%s\n' "$PORCELAIN" | sed 's/^...//' | sort)
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply .claude/hooks/checkpoint-handoff.sh:224]: Fixed in `5f04347`. Hashing the raw porcelain instead of the stripped path list, so the two status columns are part of the fingerprint and `git add` now moves it. Control: `?? new.txt` and `A  new.txt` both reduce to `new.txt` under the old `sed 's/^...//'`, so the pre-fix hash provably could not move — pinned as T10c.

Content edits inside an already-modified file still do not move it, and that is deliberate: catching those means hashing the diff on every SubagentStop, and time is the primary trigger — the next interval writes regardless of the fingerprint. The fingerprint only ever makes it write more often, never less, so the worst case stays bounded at the interval rather than being indefinite.

File: .claude/hooks/checkpoint-handoff.sh [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+SESSION_ID="${CLAUDE_SESSION_ID:-default}"
+SESSION_ID="${SESSION_ID//[^[:alnum:]_.-]/_}"
+SESSION_ID="${SESSION_ID:-default}"
+STAMP=$(date -u +%Y%m%dT%H%M%SZ 2>/dev/null)
+[[ -n "$STAMP" ]] || STAMP="unknown"
+OUT="$OUT_DIR/portable-handoff-${STAMP}-${SESSION_ID}-checkpoint.md"
+
+# Same-directory mktemp + mv is an atomic single-writer publish: a reader sees a
+# complete document or none, never a half-written one.
+if mv -f "$TMP_DOC" "$OUT" 2>/dev/null; then
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply .claude/hooks/checkpoint-handoff.sh:575]: Fixed in `5f04347`. The filename now carries the `mktemp` suffix already guaranteed unique in that directory, so two writers in the same second land on different names.

Worth noting this is not a rare interleaving in this particular script: one account usage limit fails every active session at once, which is precisely when concurrent writes happen and precisely when losing one costs the most. Pinned as T10d — two forced same-second writes must produce two distinct files and both must survive.

File: .claude/hooks/checkpoint-handoff.sh [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+    printf '.\n\n'
+    printf 'This checkpoint was written automatically rather than by a person, so what\n'
+    printf 'follows is what the repository state says, not a statement of intent. Read the\n'
+    printf 'issue linked above, if any, for the actual goal.\n\n'
+  fi
+
+  printf '## Open work\n\n'
+  if [[ -z "$PR_ROWS" ]]; then
+    printf 'No open pull requests were recorded for this account when this was written.\n'
+    printf 'Run `gh pr list --author "@me" --state open` to check the current position.\n\n'
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply .claude/hooks/checkpoint-handoff.sh:396]: Fixed in `5f04347`. The exit status of `gh` was discarded, so a failed query and a genuinely empty result were the same empty string and both rendered as "no open pull requests were recorded".

Three distinct states now — `ok` / `failed` / `skipped` — each with its own wording. This one matters beyond the immediate bug: reporting "nothing in flight" when the truth is "could not look" is the one mistake a handoff cannot afford, because the reader has no way to detect it. T10a asserts the failed path says COULD NOT BE READ and never claims none are open.

File: .claude/hooks/checkpoint-handoff.sh [hunk 4]
```diff
@@ -0,0 +1,507 @@
...
+        exit 0
+      fi
+    fi
+  fi
+fi
+
+# --- Remote state (only past the throttle, so the common path is free) -----
+
+ISSUE_NUM=""
+[[ "$BRANCH" =~ (^|[^0-9])issue-([0-9]+) ]] && ISSUE_NUM="${BASH_REMATCH[2]}"
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply .claude/hooks/checkpoint-handoff.sh]: Fixed in `5f04347`. The boundary was `[^0-9]`, so `reissue-941` matched on its trailing `e` — and any `re`/`de`/`un` prefix would do the same, publishing a link to an unrelated issue. Boundary is now `[^[:alnum:]]`. T10b pins both directions: `reissue-941` emits no issue link, `issue-941-real` still does.

**Comment by auerbachb** [comment]
@coderabbitai full review

**Comment by auerbachb** [comment]
@cursor review

**Comment by auerbachb** [comment]
@graphite-app re-review

File: .claude/hooks/checkpoint-handoff.sh [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+  printf "## What we're working on\n\n"
+  if (( minimal )); then
+    printf 'This checkpoint was written automatically by a background task, so it records\n'
+    printf 'no statement of intent. The repository is `%s`. Run `git log --oneline -20`\n' "$REPO_SLUG"
+    printf 'and `git status` in the working directory below to see what was underway.\n\n'
+  else
+    printf 'Repository `%s`' "$REPO_SLUG"
+    if [[ -n "$ISSUE_NUM" ]]; then
+      printf ', working on issue %s' "$ISSUE_NUM"
+      [[ -n "$REPO_SLUG" ]] && printf ' (https://github.com/%s/issues/%s)' "$REPO_SLUG" "$ISSUE_NUM"
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply .claude/hooks/checkpoint-handoff.sh]: Fixed in bc37e4d after rebasing onto current main. Remote lookup now requires a real Git checkout, so a basename fallback cannot authorize gh to use unrelated ambient/default repository context. Issue URLs are emitted only for an owner/repository-shaped slug; a basename remains a display label only.

T10e pins both sides: a stub gh outside a checkout cannot leak PR #42 into the handoff, and an issue-941-real branch in a no-origin repository can name issue 941 without generating a malformed GitHub URL. The checkpoint suite passes all 63 assertions.
